### ISSUE TITLE
[BUG] New silence settings rejecting training files that were previously valid

### ISSUE THREAD

**Issue by mikeoliphant** [sdatkinson/neural-amp-modeler#268]
The new silence check is now rejecting custom v1_1_1.wav training files (training files with additional training data added) that previously worked.

Maybe consider only enforcing the silence constraint for v2 training files?

**Comment by vossenv** [comment]
You can click "ignore checks" for this can't you?

**Comment by mikeoliphant** [comment]
> You can click "ignore checks" for this can't you?

I'm using the colab notebook.

**Comment by mikeoliphant** [comment]
> I'm using the colab notebook.

Nevermind - I see the parameter now.

**Comment by mikeoliphant** [comment]
Unfortunately, passing "ignore_checks=True" to run() does not skip the silence check.

**Comment by sdatkinson** [comment]
Looking into this

**Comment by sdatkinson** [comment]
Here's the test that asserts that the "stock" reamp files (`v1.wav`, `v1_1_1.wav`, and `v2_0_0.wav`) are valid: https://github.com/sdatkinson/neural-amp-modeler/blob/b65d557811939f55d9ed6af2462e6c6ee7931273/tests/test_nam/test_train/test_core.py#L126

The requirement is that the 0.4 seconds (aka [`_DEFAULT_REQUIRE_INPUT_PRE_SILENCE`](https://github.com/sdatkinson/neural-amp-modeler/blob/b65d557811939f55d9ed6af2462e6c6ee7931273/tests/test_nam/test_train/test_core.py#LL134C21-L134C55), defined [here](https://github.com/sdatkinson/neural-amp-modeler/blob/b65d557811939f55d9ed6af2462e6c6ee7931273/nam/data.py#L212)) be silent before the start of the validation segment.

The test is skipped during CI because it requires the data files, and those aren't downloaded during. But it works on my computer. Can you confirm that you see the same? (You need to download [`v1_1_1.wav`](https://drive.google.com/file/d/1v2xFXeQ9W2Ks05XrqsMCs2viQcKPAwBk/view?usp=drive_link) to `./tests/resources/v1_1_1.wav`).

**First question: can you confirm that the test passes for you?**

**Second question: Does it work with the stock training signal?**

The validation segments starts 9 seconds before the end (aka `-432_000` samples). So if you put your extra training data _right_ before that, it might fail if your new data don't end in at least 0.4 seconds of silence (so that the response can chill out before the validation signal starts).

Does that sound like the culprit?

Over here, if I download the `v1_1_1.wav` and `output.wav` fresh from the links in the README, it works fine on local GUI (and probably easy Colab, but I haven't checked yet). **Third question: Can you replicate this?**

**Comment by sdatkinson** [comment]
> if "ignore_checks" means "all bets are off", it should probably still let this pass.

Maaaybe. Given how holy "ESR" seems to be by users, then I should probably not let them compute ESR on a different validation set than "the" validation set?

**Comment by mikeoliphant** [comment]
> Maaaybe. Given how holy "ESR" seems to be by users, then I should probably not let them compute ESR on a different validation set than "the" validation set?

Yeah, I've been dealing with the "what does relative ESR even mean" issue today, so I can't say that I disagree with you in principle.

I'm going to close this for now - it be given more thought if it comes up as a more pervasive issue with people trying to use older captures.
